shop/src/main/java/ro/msg/learning/shop/service/strategy/MostAbundantLocationStrategy.java

```diff
@@ -0,0 +1,43 @@
+package ro.msg.learning.shop.service.strategy;
+
+import ro.msg.learning.shop.model.OrderDetail;
+import ro.msg.learning.shop.model.Stock;
+import ro.msg.learning.shop.repository.StockRepository;
+import ro.msg.learning.shop.service.exception.LocationStockException;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class MostAbundantLocationStrategy implements OrderStrategy {
+    private static final String UNFOUND_STOCK = "No stock with a suitable location!";
+    private final StockRepository stockRepository;
+
+    public MostAbundantLocationStrategy(StockRepository stockRepository) {
```

Comment:
Why do you need this constructor? The stock repository could be injected with `@RequiredArgsConstructor`. Also, as I mentioned before only the service can speak with the repository of an entity